keanu-python/keanu/vartypes.py

```diff
@@ -1,16 +1,65 @@
+from typing import Iterable, Union, Type, Any, TYPE_CHECKING
 import numpy as np
-import pandas as pd
+from pandas import Series, DataFrame
+from keanu.vertex.ops import VertexOps
+from .base import JavaObjectWrapper
 
-int_types = (int, np.integer)
 
-float_types = (float, np.float)
+# see numpy's scalar hierarchy: https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html#scalars
+int_types = Union[int, np.integer]
+float_types = Union[float, np.floating]
+bool_types = Union[bool, np.bool_]
 
-bool_types = (bool, np.bool_)
+primitive_types = Union[int_types, float_types, bool_types]
 
-primitive_types = int_types + float_types + bool_types
+'''
+Tensor arg types
+'''
+pandas_types = Union[Series, DataFrame]
+numpy_types = Union[np.ndarray]
+pandas_and_numpy_types = Union[pandas_types, numpy_types]
 
-pandas_types = (pd.Series, pd.DataFrame)
+# mypy treats bool as a subtype of int and int as a subtype of float
+# see : https://github.com/python/mypy/issues/1850
+tensor_arg_types = Union[primitive_types, pandas_and_numpy_types]
+bool_tensor_arg_types = Union[bool_types, pandas_and_numpy_types]
+int_and_bool_tensor_arg_types = Union[int_types, bool_types, pandas_and_numpy_types]
 
-numpy_types = (np.ndarray, )
+'''
+Vertex arg types
+'''
+wrapped_java_types = Union[JavaObjectWrapper]
+shape_types = Iterable[primitive_types]
 
-const_arg_types = numpy_types + pandas_types + primitive_types
+# mypy treats bool as a subtype of int and int as a subtype of float
+# see : https://github.com/python/mypy/issues/1850
+vertex_param_types = Union[tensor_arg_types, wrapped_java_types, VertexOps]
+bool_vertex_param_types = Union[bool_tensor_arg_types, wrapped_java_types, VertexOps]
+int_and_bool_vertex_param_types = Union[int_and_bool_tensor_arg_types, wrapped_java_types, VertexOps]
+
+vertex_arg_types = Union[vertex_param_types, shape_types]
```

Comment:
This is only used in 2 places, and it's easy to confuse with `vertex_param_types` - so I'd suggest inlining it for clarity.